### ISSUE TITLE
System.Text.Json support

### ISSUE THREAD

**Issue by BigDaddy1337** [alexanderkozlenko/json-rpc#116]
Are there any plans for migration or support for System.Text.Json (for example, backward compatibility support for Json.NET with the choice of serializer through settings)? In order to remove the Json.NET dependency, increase throughput and reduce memory allocation?

**Comment by alexanderkozlenko** [comment]
Yes, I'm planning to replace Newtonsoft.Json with System.Text.Json by the .NET 5 release date.

**Comment by BigDaddy1337** [comment]
Hi @alexanderkozlenko, do you have any updates about support for System.Text.Json?

**Comment by alexanderkozlenko** [comment]
Hi @BigDaddy1337, the main reason the library has not been migrated to System.Text.Json yet is the absence of JsonStreamReader/JsonStreamWriter implementations in .NET 5.0 (see [System.Text.Json roadmap](https://github.com/dotnet/runtime/tree/release/5.0/src/libraries/System.Text.Json/roadmap)). A custom implementation will not be a trivial one, and it makes less sense to do it while it is already on the roadmap. Thus, I'm not planning this migration at least till .NET 6 release.

If you have any thoughts or suggestions on this matter, I will be glad to discuss them, since the migration to System.Text.Json will be a great benefit.
